### PR TITLE
flesh out some further deployment integration

### PR DESCRIPTION
**Pull request by grahamc**
##### Checklist

- [ ] Formatted with `cargo fmt`
- [ ] Built with `nix build`
- [ ] Ran flake checks with `nix flake check`
- [ ] Added or updated relevant tests (leave unchecked if not applicable)
- [ ] Added or updated relevant documentation (leave unchecked if not applicable)
- [ ] Linked to related issues (leave unchecked if not applicable)

##### Validating with `install.determinate.systems`

If a maintainer has added the `upload to s3` label to this PR, it will become available for installation via `install.determinate.systems`:

```shell
curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix/pr/$PR_NUMBER | sh -s -- install
```

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -20,7 +20,8 @@
     };
 
     determinate = {
-      url = "https://flakehub.com/f/DeterminateSystems/determinate/0.1.tar.gz";
+      #url = "https://flakehub.com/f/DeterminateSystems/determinate/0.1.tar.gz";
+      url = "path:///Users/grahamchristensen/src/github.com/DeterminateSystems/determinate";
```
Comment (grahamc) [inline flake.nix:24]: pending nixd 63 and determinate 18

File: flake.nix [hunk 2]
```diff
@@ -106,7 +106,7 @@
             cargoTestOptions = f: f ++ [ "--all" ];
 
             NIX_INSTALLER_TARBALL_PATH = nixTarballs.${final.stdenv.system};
-            DETERMINATE_NIXD_BINARY_PATH = if final.stdenv.system == "x86_64-linux" || final.stdenv.system == "aarch64-linux" then "${inputs.determinate.packages.${final.stdenv.system}.default}/bin/determinate-nixd" else null;
+            DETERMINATE_NIXD_BINARY_PATH = if builtins.elem final.stdenv.system [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin" ] then "${inputs.determinate.packages.${final.stdenv.system}.default}/bin/determinate-nixd" else null;
```
Comment (cole-h) [inline flake.nix:109]: We should move this list of systems into a variable.

File: src/action/common/provision_determinate_nixd.rs [hunk 3]
```diff
@@ -4,20 +4,21 @@ use std::path::PathBuf;
...
 
-use crate::action::common::configure_determinate_nixd_init_service::DETERMINATE_NIXD_SERVICE_SRC;
 use crate::action::{
     Action, ActionDescription, ActionError, ActionErrorKind, ActionTag, StatefulAction,
 };
 
+#[cfg(target_os = "linux")]
 const DETERMINATE_NIXD_BINARY_PATH: &str = "/nix/determinate/determinate-nixd";
+#[cfg(target_os = "macos")]
+const DETERMINATE_NIXD_BINARY_PATH: &str = "/usr/local/bin/determinate-nixd";
```
Comment (cole-h) [inline src/action/common/provision_determinate_nixd.rs:14]: nit: you've previously complained about how cfg-ing things makes it harder to develop with -- could we not cfg these and instead match on platform like we do in other places?

File: src/settings.rs [hunk 4]
```diff
@@ -23,8 +23,7 @@ pub const NIX_TARBALL: &[u8] = include_bytes!(env!("NIX_INSTALLER_TARBALL_PATH")
 
 #[cfg(all(
     feature = "determinate-nix",
-    target_os = "linux",
-    not(target_arch = "x86")
+    not(all(target_os = "linux", target_arch = "x86"))
```
Comment (cole-h) [inline src/settings.rs:26]: Can we get a comment about this cfg? "If determinate-nix is enabled, and the target is not i686-linux, then embed the binary"

(It took me like 5 tries to understand what was going on here)

**Review by lucperkins — APPROVED** [review]

File: src/settings.rs [hunk 5]
```diff
@@ -34,8 +33,7 @@ pub const DETERMINATE_NIXD_BINARY: Option<&[u8]> =
 
 #[cfg(not(all(
     feature = "determinate-nix",
-    target_os = "linux",
-    not(target_arch = "x86")
+    not(all(target_os = "linux", target_arch = "x86"))
```
Comment (cole-h) [inline src/settings.rs:36]: Same here: "If determinate-nix is not enabled, or the target is i686-linux, do not embed the binary"

File: flake.nix [hunk 6]
```diff
@@ -177,7 +177,7 @@
               darwin.apple_sdk.frameworks.SystemConfiguration
             ])
             ++ lib.optionals (pkgs.stdenv.isLinux) (with pkgs; [
-              checkpolicy
+              # checkpolicy # failed to build 2024-08-08
```
Comment (cole-h) [inline flake.nix:180]: Can we find a Nixpkgs that it does build with, or alternatively not update Nixpkgs?

**Review by cole-h — CHANGES_REQUESTED** [review]

File: flake.nix [hunk 7]
```diff
@@ -73,6 +73,8 @@
       nixTarballs = forAllSystems ({ system, ... }:
         inputs.nix.tarballs_direct.${system}
           or "${inputs.nix.checks."${system}".binaryTarball}/nix-${inputs.nix.packages."${system}".default.version}-${system}.tar.xz");
+
+      optionalPathToDeterminateNixd = system: if builtins.elem system [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin" ] then "${inputs.determinate.packages.${system}.default}/bin/determinate-nixd" else null;
```
Comment (cole-h) [inline flake.nix:77]: Better, but I really did mean to split out the list:
```suggestion
      supportedDeterminateNixdSystems = [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin" ];
      optionalPathToDeterminateNixd = system: if builtins.elem system supportedDeterminateNixdSystems then "${inputs.determinate.packages.${system}.default}/bin/determinate-nixd" else null;
```

Comment (grahamc) [reply flake.nix]: Done!

**Review by cole-h — APPROVED** [review]

**Pull request merged**